nerfstudio/data/datamanagers/base_datamanager.py

```diff
@@ -426,22 +440,44 @@ def __init__(
 
         super().__init__()
 
-    def create_train_dataset(self) -> InputDataset:
+    def get_dataset_type(self) -> Type[TDataset]:
+        """Returns the dataset type passed as the generic argument"""
+        orig_class = get_orig_class(self, default_to__class__=True)
+        if orig_class is VanillaDataManager:
+            return TDataset.__default__  # typing: ignore
+        if hasattr(orig_class, "__origin__") and orig_class.__origin__ is VanillaDataManager:
+            return orig_class.__args__[0]
+        # For inherited classes, we need to find the correct type to instantiate
+        for base in getattr(self, "__orig_bases__", []):
+            if hasattr(base, "__origin__") and base.__origin__ is VanillaDataManager:
+                for value in base.__args__:
+                    if isinstance(value, ForwardRef):
```

Comment:
do we actually use forward refs, or is this just being thorough? if it's the latter I would vote to just add an `assert not isinstance(value, ForwardRef)` to keep things simple

Reply:
Currently the code does not use ForwardRefs. I will drop the support.

Reply:
Thanks!